addon/factory-guy.js

```diff
@@ -79,6 +80,15 @@ var FactoryGuy = function () {
   this.getFixtureBuilder = function () {
     return fixtureBuilder;
   };
+  this.getFixtureBuilderFor = function (modelName) {
+    if (!fixtureBuilders[modelName]) {
+      var factory = new FixtureBuilderFactory(store, modelName);
+
+      fixtureBuilders[modelName]  = factory.getFixtureBuilder();
+    }
+
+    return fixtureBuilders[modelName];
```

Comment:
A better map could be to have an instance per type instead of model, so if the adapter is json it returns something like `fixtureBuilder['JSONAdapter']`